zebra-chain/src/transaction.rs

```diff
@@ -804,6 +860,10 @@ impl Transaction {
                 sapling_shielded_data: Some(sapling_shielded_data),
                 ..
             } => Box::new(sapling_shielded_data.spends_per_anchor()),
+            Transaction::V6 {
+                sapling_shielded_data: Some(sapling_shielded_data),
+                ..
+            } => Box::new(sapling_shielded_data.spends_per_anchor()),
```

Comment:
It would be a good simplification, but it's probably not compatible with the "tx-v6" feature flag. I’ve provided more details in [this response](https://github.com/QED-it/zebra/pull/16#discussion_r1835492766).